extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQueryEngine.java

```diff
@@ -173,10 +175,12 @@ public ScanResultValue next()
                                 ScanQueryRunnerFactory.CTX_COUNT,
                                 (long) responseContext.get(ScanQueryRunnerFactory.CTX_COUNT) + (offset - lastOffset)
                             );
-                            responseContext.put(
-                                ScanQueryRunnerFactory.CTX_TIMEOUT_AT,
-                                timeoutAt - (System.currentTimeMillis() - start)
-                            );
+                            if (hasTimeout) {
+                              responseContext.put(
```

Comment:
Maybe it's not that easy to fix this, because `CTX_TIMEOUT_AT` and `CTX_COUNT` are used in some non-obvious way in `ScanQueryEngine`